modular_skyrat/master_files/code/modules/clothing/towels.dm

```diff
@@ -0,0 +1,522 @@
+	/// The shape we're currently in.
+	var/shape = TOWEL_FOLDED
+	/// How many units of liquid can this towel store?
+	var/max_reagent_volume = 25
+	/// Are we currently wet?
+	var/wet = FALSE
+
+
+/obj/item/towel/Initialize(mapload)
+	. = ..()
+
+	create_reagents(max_reagent_volume)
+	AddComponent(/datum/component/liquids_interaction, TYPE_PROC_REF(/obj/item/towel, attack_on_liquids_turf))
+	AddComponent(/datum/component/surgery_initiator) // Since you can do it with bedsheets, why not with towels too?
+
+	register_context()
+	register_item_context()
+
+
+/obj/item/towel/examine(mob/user)
+	. = ..()
+
+	if(wet)
+		. += span_notice("\nIt appears to be wet.")
+
+
+	if(!ishuman(user) && !iscyborg(user))
+		return
+
+	. += "" // Just for an empty line
+
+	var/in_hands = TRUE
+	if(ishuman(user))
+		in_hands = user.get_active_held_item() == src || user.get_inactive_held_item() == src
+
+		if(in_hands)
+			. += span_notice("<b>Use in hand</b> to shape [src] into something different.")
+
+	if(in_hands && shape != TOWEL_FOLDED)
+		. += span_notice("<b>Ctrl-click</b> to [wet && ishuman(user) ? "wring parts of the liquids out of [src]" : "fold [src] neatly"].")
+
+	if(iscyborg(user))
+		return
+
+	if(shape == TOWEL_FULL || shape == TOWEL_WAIST)
+		. += span_notice("<b>Alt-click</b> to adjust the fit of [src].")
+
+	if(wet)
+		. += span_notice("<b>Right-click</b> [src] on a bucket to wring the liquids out of it and transfer a portion of them to the bucket.")
+		. += span_notice("<b>Wash in a washing machine</b> in order to clean [src].")
+
+
+/obj/item/towel/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	if(istype(held_item) && (held_item.tool_behaviour == TOOL_WIRECUTTER || held_item.get_sharpness()) && !(flags_1 & HOLOGRAM_1))
+		context[SCREENTIP_CONTEXT_LMB] = "Shred into cloth"
+
+	if(ishuman(user))
+		if((shape == TOWEL_FULL || shape == TOWEL_WAIST))
+			context[SCREENTIP_CONTEXT_ALT_LMB] = "Adjust Fit"
+
+		var/mob/living/carbon/human/towel_user = user
+		var/worn = towel_user.wear_suit == src || towel_user.head == src
+
+		if(!worn)
+			context[SCREENTIP_CONTEXT_LMB] = "Change Shape"
+			context[SCREENTIP_CONTEXT_CTRL_LMB] = wet ? "Wring" : "Fold"
+
+	if(iscyborg(user))
+		context[SCREENTIP_CONTEXT_CTRL_LMB] = wet ? "Wring" : "Fold"
+
+	return CONTEXTUAL_SCREENTIP_SET
+
+
+/obj/item/towel/add_item_context(datum/source, list/context, mob/living/target)
+	if(isliving(target) && target.fire_stacks < 0) // If the target indeed is a living mob, and has wet stacks (which are just negative fire stacks)
+		context[SCREENTIP_CONTEXT_LMB] = "Dry up"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	return NONE
+
+
+/obj/item/towel/attack(mob/living/target_mob, mob/living/user, params)
+	. = ..()
+	if(!user.Adjacent(target_mob))
+		return
+
+
+	var/free_space = reagents.maximum_volume - reagents.total_volume
+	if(free_space <= 0)
+		to_chat(user, span_warning("Your [src] can't absorb any more liquid!"))
```

Comment:
It would be nice to see lots of these attack to_chats become balloon_alerts but my brain doesnt work right now so I can't suggest good ideas, and I can't blame you if you don't want to or if you have the same issue

I just hate to_chats... grr....